great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -0,0 +1,447 @@
+        DataContextConfigDefaults.EXPECTATIONS_BASE_DIRECTORY.value,
+        DataContextConfigDefaults.PLUGINS_BASE_DIRECTORY.value,
+        DataContextConfigDefaults.PROFILERS_BASE_DIRECTORY.value,
+        GX_UNCOMMITTED_DIR,
+    ]
+    GX_DIR = "great_expectations"
+    GX_YML = "great_expectations.yml"
+    GX_EDIT_NOTEBOOK_DIR = GX_UNCOMMITTED_DIR
+    DOLLAR_SIGN_ESCAPE_STRING = r"\$"
+
+    def __init__(
+        self, context_root_dir: str, runtime_environment: Optional[dict] = None
+    ) -> None:
+        self._context_root_directory = context_root_dir
+        super().__init__(runtime_environment=runtime_environment)
+
+    def _init_datasource_store(self):
+        pass
+
+    def _init_variables(self):
+        pass
+
+    def _save_project_config(self) -> None:
+        """
+        See parent 'AbstractDataContext._save_project_config()` for more information.
+
+        Explicitly override base class implementation to retain legacy behavior.
+        """
+        logger.debug("Starting DataContext._save_project_config")
+
+        config_filepath = os.path.join(self.root_directory, self.GX_YML)  # type: ignore[arg-type]
+
+        try:
+            with open(config_filepath, "w") as outfile:
+                self.config.to_yaml(outfile)
+        except PermissionError as e:
+            logger.warning(f"Could not save project config to disk: {e}")
+
+    def _check_for_usage_stats_sync(self, project_config: DataContextConfig) -> bool:
+        """
+        If there are differences between the DataContextConfig used to instantiate
+        the DataContext and the DataContextConfig assigned to `self.config`, we want
+        to save those changes to disk so that subsequent instantiations will utilize
+        the same values.
+
+        A small caveat is that if that difference stems from a global override (env var
+        or conf file), we don't want to write to disk. This is due to the fact that
+        those mechanisms allow for dynamic values and saving them will make them static.
+
+        Args:
+            project_config: The DataContextConfig used to instantiate the DataContext.
+
+        Returns:
+            A boolean signifying whether or not the current DataContext's config needs
+            to be persisted in order to recognize changes made to usage statistics.
+        """
+        project_config_usage_stats: Optional[
+            AnonymizedUsageStatisticsConfig
+        ] = project_config.anonymous_usage_statistics
+        context_config_usage_stats: Optional[
+            AnonymizedUsageStatisticsConfig
+        ] = self.config.anonymous_usage_statistics
+
+        if (
+            project_config_usage_stats.enabled is False  # type: ignore[union-attr]
+            or context_config_usage_stats.enabled is False  # type: ignore[union-attr]
+        ):
+            return False
+
+        if project_config_usage_stats.explicit_id is False:  # type: ignore[union-attr]
```

Comment:
Can we fix these issues on `AnonymizedUsageStatisticsConfig` + `AnonymizedUsageStatisticsConfig`?

Reply:
We've got a ticket in the context epic to resolve `mypy` issues - mind if we tackle that there?

Reply:
That works for me.